src/Auth/JwtAuthDriver.php

```diff
@@ -82,17 +82,25 @@ public function __construct(
      */
     public function user()
     {
-        $this->validate();
+        if (!$this->validate()) {
+            return null;
+        }
+
+        if (!is_null($this->user)) {
+            return $this->user;
+        }
 
-        if ($user = $this->provider->retrieveById($this->jwtPayload->sub)) {
-            return $user;
+        // retrieve user from db or create a new user by id
+        if (is_null($user = $this->provider->retrieveById($this->jwtPayload->sub))) {
```

Comment:
It might never happen, but as a safe procedure:
```suggestion
        if (is_null($user = $this->provider->retrieveById($this->jwtPayload?->sub))) {
```